mythril/mythril/mythril_analyzer.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import logging
+import traceback
+from typing import Optional, List
+
+from . import MythrilDisassembler
+from mythril.support.source_support import Source
+from mythril.support.loader import DynLoader
+from mythril.analysis.symbolic import SymExecWrapper
+from mythril.analysis.callgraph import generate_graph
+from mythril.analysis.traceexplore import get_serializable_statespace
+from mythril.analysis.security import fire_lasers, retrieve_callback_issues
+from mythril.analysis.report import Report, Issue
+from mythril.ethereum.evmcontract import EVMContract
+
+log = logging.getLogger(__name__)
+
+
+class MythrilAnalyzer:
+    """
+    The Mythril Analyzer class
+    Responsible for the analysis of the smart contracts
+    """
+
+    def __init__(
+        self,
+        disassembler: MythrilDisassembler,
+        requires_dynld: bool = False,
+        onchain_storage_access: bool = True,
+    ):
+        """
+
+        :param disassembler: The MythrilDisassembler class
+        :param requires_dynld: whether dynamic loading should be done or not
+        :param onchain_storage_access: Whether onchain access should be done or not
+        """
+        self.eth = disassembler.eth
+        self.contracts = disassembler.contracts or []  # type: List[EVMContract]
+        self.enable_online_lookup = disassembler.enable_online_lookup
+        self.dynld = requires_dynld
+        self.onchain_storage_access = onchain_storage_access
+
+    def dump_statespace(
```

Comment:
Do we use this functionality for something?

Reply:
For getting the statespace in json format